### ISSUE TITLE
AbstractColumn.asStringColumn() does not convert missing values

### ISSUE THREAD

**Issue by ccleva** [jtablesaw/tablesaw#597]
Hi,

Similar to issue #577, asStringColumn() does not convert missing values.

For Double/Float columns the result is the string 'NaN', for Date/Time columns the result is the string 'null'. Both are acceptable, but for Int/Short/Long columns the result is String.valueOf(Integer/Short/Long MIN_VALUE) which is misleading.

I can provide a fix, but what would be your preferred solution:
- Consistently use the string 'NaN' for missing numbers and the string 'null' for missing objects
- Use StringColumnType.missingValueIndicator() for all missing values

My preferred approach would be to delegate string representation to the column itself by using getUnformattedString() (which uses String.valueOf for regular values, so not a breaking change). 

We could also add a new asFormattedStringColumn() method that would use getString() for completeness.

Let me know what you think.
Thanks

**Comment by lwhite1** [comment]
Hi @ccleva,

thanks for the report, and especially for the offer to fix. 

assuming that the existing code is enumerating the values in the column, and assuming the columns still implement a missing value test (I'm not sure they do) i would attempt to do something like 

```
if (theSourceColumn.isMissingValue(x)) {
   theStringColumn.appendMissing();
} else { whatever it does now }
```
it's possible you'd have to do this:

```
if (theSourceColumn.type().isMissingValue(x)... 
```

**Comment by ccleva** [comment]
Thanks for the reply @lwhite1.

What you describe was my initial idea, but the valueIsMissing(x) methods are deprecated and the ColumnType class does not have a isMissingValue(x) method.

This means either every Column class needs to have its own implementation of asStringColumn or AbstractColumn.asStringColumn has to use isMissing(i) and get(i) in the loop, which means accessing the value twice and a hit in performance:

```
    	if (theSourceColumn.isMissing(i)) {
    		theStringColumn.appendMissing();
    	} else {
    		theStringColumn.append(String.valueOf(theSourceColumn.get(i));
    	}
```
I implemented a fix by using delegation (PR #599) to the column getUnformattedString method. It is in practice equivalent to what you describe but allows to keep asStringColumn implementation in the AbstractColumn class.

Let me know if this is ok.

**Comment by lwhite1** [comment]
fixed. thanks!
